### PR TITLE
agi: init at 1.1.0-dev-20210413

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Building this has proven to be a bit of a pain, so for now I'm just patching the binary.

Testing is much appreciated.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/agi/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    done
+    cp -r ${makeDesktopItem {
+      name = pname;
+      desktopName = "Android GPU Inspector";
+      exec = "agi";
+      icon = "agi";
+      type = "Application";
+      categories = "Development;Debugger;Graphics;3DGraphics";
+      terminal = "false";
+    }}/share/applications $out/share
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/agi/default.nix:56]: Please use copyDesktopItems.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/graphics/agi/default.nix (hunk 1 above)
Comment (IvarWithoutBones) [reply pkgs/tools/graphics/agi/default.nix:56]: I'm not very familiar with this function, but from grepping through nixpkgs i gather you should have an `let` block like this:
```nix
let
  desktopItem = makeDesktopItem {
  };
in stdenv.mkDerivation rec {
```

Which IMO looks way worse, is there any guidelines on this? If no, I'd prefer to leave it as-is.

Comment (SuperSandro2000) [reply pkgs/tools/graphics/agi/default.nix:58]: You can also inline it like https://github.com/NixOS/nixpkgs/blob/7248f89da2dab8bca3a9f4f11d0407d0c8e25e89/pkgs/applications/blockchains/mycrypto/default.nix#L20.

Comment (IvarWithoutBones) [reply pkgs/tools/graphics/agi/default.nix:58]: Sure, that makes sense.

I have not been able to get the function working, as with everything I've tried it simply ends up not copying over the desktop item.

My WIP derivation using `copyDesktopItems`: https://gist.github.com/IvarWithoutBones/7671f660157b6ba8206a5f787122594f
```
~/nix/nixpkgs/pkgs/tools/graphics/agi > ls ./result/share/
icons
```

Comment (SuperSandro2000) [reply pkgs/tools/graphics/agi/default.nix:58]: You are not inheriting desktopItem. https://github.com/NixOS/nixpkgs/blob/ffd29c76f682ffdec5e9a2b7fb81c3f07f679361/pkgs/applications/misc/dbeaver/default.nix#L74

Comment (IvarWithoutBones) [reply pkgs/tools/graphics/agi/default.nix]: I've followed the example you posted, as I much prefer this over the `let ... in` block.

Done :+1:

**Pull request merged**